lib/ansible/modules/crypto/openssl_certificate.py

```diff
@@ -0,0 +1,766 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2016-2017, Yanis Guenane <yanis+ansible@guenane.org>
+# (c) 2017, Markus Teufelberger <mteufelberger+ansible@mgit.at>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: openssl_certificate
+author:
+  - Yanis Guenane (@Spredzy)
+  - Markus Teufelberger (@MarkusTeufelberger)
+version_added: "2.4"
+short_description: Generate and/or check OpenSSL certificates
+description:
+    - "This module allows one to (re)generate OpenSSL certificates. It implements a notion
+       of provider (ie. 'selfsigned', 'acme', 'assertonly') for your certificate.
+       The 'assertonly' provider is intended for use cases where one is only interested in
+       checking properties of a supplied certificate.
+       Many properties that can be specified in this module are for validation of an
+       existing or newly generated certificate. The proper place to specify them, if you
+       want to receive a certificate with these properties is a CSR (Certificate Signing Request).
+       It uses the pyOpenSSL python library to interact with OpenSSL."
+requirements:
+    - "python-pyOpenSSL >= 0.15"
```

Comment:
`PyOpenSSL>= 0.15` seems only required for `SelfSignedCertificate`, with other providers older PyOpenSSL versions should work fine: this information could be added here.